### PR TITLE
[Part II] Add shortened links CRUD

### PR DESCRIPTION
**Pull request by brunvez**

### CODE REVIEW

File: app/javascript/controllers/shortened_link_form_controller.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  validate(e) {
+    this.stimulate('ShortenedLinkFormReflex#validate', [e.target], this.shortenedLinkParams)
+  }
+
+  save(e) {
+    this.stimulate('ShortenedLinkFormReflex#save', [e.target], this.shortenedLinkParams)
+  }
+
+  destroy
```
Comment (brunvez) [inline app/javascript/controllers/shortened_link_form_controller.js:33]: Remove

Comment (santib) [inline app/javascript/controllers/shortened_link_form_controller.js:33]: what's this?

File: app/javascript/controllers/shortened_link_form_controller.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { Controller } from "stimulus"
+import StimulusReflex from 'stimulus_reflex'
+
+export default class extends Controller {
+  static targets = ['name', 'shortenedPath', 'originalUrl']
+
+  connect () {
```
Comment (horacio) [inline app/javascript/controllers/shortened_link_form_controller.js:7]:
```suggestion
  connect() {
```

File: app/javascript/controllers/shortened_links_controller.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { Controller } from "stimulus"
+import StimulusReflex from 'stimulus_reflex'
+
+export default class extends Controller {
+  connect () {
```
Comment (horacio) [inline app/javascript/controllers/shortened_links_controller.js:5]:
```suggestion
  connect() {
```

File: app/javascript/controllers/shortened_link_form_controller.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  save(e) {
+    this.stimulate('ShortenedLinkFormReflex#save', [e.target], this.shortenedLinkParams)
+  }
+
+  destroy
+
+  get shortenedLinkId() {
+    const id = parseInt(this.data.get('linkId'))
```
Comment (juanmanuelramallo) [inline app/javascript/controllers/shortened_link_form_controller.js:36]:
```suggestion
    const id = parseInt(this.data.get('linkId'), 10)
```

File: app/javascript/controllers/shortened_link_form_controller.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    return !!id ? id : null
```
Comment (juanmanuelramallo) [inline app/javascript/controllers/shortened_link_form_controller.js:37]:
```suggestion
    return isNan(id) ? null : id
```

File: app/models/shortened_link.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+class ShortenedLink < ApplicationRecord
+  validates :name, :shortened_path, :original_url, presence: true
+  validates :shortened_path, uniqueness: true
```
Comment (juanmanuelramallo) [inline app/models/shortened_link.rb:3]: Wouldn't this `shortened_path` attribute potentially clash with a `shortened_path` route defined in the routes file? Same for `original_url`

File: app/views/shortened_links/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,70 @@
+<div class="live-modal"
+     data-controller="shortened-link-form"
+     data-action="keydown->shortened-link-form#handleKeyDown"
+     data-shortened-link-form-link-id="<%= shortened_link.id %>"
+>
+  <div class="live-modal-backdrop" data-action="click->shortened-link-form#closeModal">
+    <div class="live-modal-content" data-action="click->shortened-link-form#trapClick">
+      <div class="live-modal-close" data-action="click->shortened-link-form#closeModal">&times;</div>
+      <% if form_action == :edit %>
```
Comment (santib) [inline app/views/shortened_links/_form.html.erb:9]: I thought this needed to have the `@` like `@form_action` 🤔

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -6,10 +6,13 @@
...
 
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css">
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
+  <div class="container">
     <%= yield %>
+  </div>
```
Comment (juanmanuelramallo) [inline app/views/layouts/application.html.erb:16]:
```suggestion
    <div class="container">
      <%= yield %>
    </div>
```

File: app/javascript/controllers/shortened_link_form_controller.js (hunk 4 above)
Comment (horacio) [reply app/javascript/controllers/shortened_link_form_controller.js:36]: Isn't 10 the default value?

File: app/models/shortened_link.rb (hunk 6 above)
Comment (brunvez) [reply app/models/shortened_link.rb:3]: Could be a case, but I don't think anybody would define a route called `/shortened` or `/original`. Still open to change for better names though I think these are clear for the reader

File: app/views/shortened_links/_form.html.erb (hunk 7 above)
Comment (brunvez) [reply app/views/shortened_links/_form.html.erb:9]: I sent it as a local 😄

File: app/javascript/controllers/shortened_link_form_controller.js (hunk 4 above)
Comment (juanmanuelramallo) [reply app/javascript/controllers/shortened_link_form_controller.js:36]: it's actually derived from the string passed in the first param

Comment (brunvez) [reply app/javascript/controllers/shortened_link_form_controller.js:36]: Never trust Javascript

**Pull request merged**